recipes/powershell_dev.rb

```diff
@@ -16,7 +16,7 @@
 # limitations under the License.
 
 powershell_script 'ExecutionPolicyUnrestricted' do
-  architecture :x86_64
+  architecture ENV["PROCESSOR_ARCHITECTURE"] == "x86" ? :i386 : :x86_64
```

Comment:
Actually, just leaving `architecture` unspecified results in that behavior. Removing that line altogether should fix it, though it's less readable, so replacing with a comment would be good.

Also, the next `powershell_script` instance is then unnecessary, since it explicitly does the 32-bit stuff, which will have been done by the block above. 

To avoid the duplication, either one of these blocks could have an extra expression in the guard, something like `$env:processor_architecture -eq 'AMD64'` so that only one of those blocks executes on a 32-bit system. For readability, you could leave the existing `x86_64` specification in, but add a guard to it -- that will work as long as we check that in the provider and not in the resource.